ignite/services/plugin/cache.go

```diff
@@ -0,0 +1,143 @@
+package plugin
+
+import (
+	"encoding/gob"
+	"fmt"
+	"net"
+	"path"
+	"path/filepath"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	chainconfig "github.com/ignite/cli/ignite/config"
+	"github.com/ignite/cli/ignite/pkg/cache"
+)
+
+const (
+	cacheFileName  = "ignite_plugin_cache.db"
+	cacheNamespace = "plugin.rpc.context"
+)
+
+var (
+	storage      *cache.Storage
+	storageCache *cache.Cache[ConfigContext]
+)
+
+func init() {
+	gob.Register(hplugin.ReattachConfig{})
+	gob.Register(net.UnixAddr{})
+}
+
+type ConfigContext struct {
+	Plugin hplugin.ReattachConfig
+	Addr   net.UnixAddr
+}
+
+func WritePluginConfigCache(pluginPath string, conf hplugin.ReattachConfig) error {
+	name := path.Base(pluginPath)
+
+	if name == "." {
+		return fmt.Errorf("provided path is invalid: %s", pluginPath)
+	}
+
+	if conf.Addr == nil {
+		return fmt.Errorf("plugin Address info cannot be empty")
+	}
+
+	confCont := ConfigContext{}
+	// TODO: figure out a better way of resolving the type of network connection is established between plugin server and host
+	// currently this will always be a unix network socket. but this might not be the case moving forward.
+	ua, err := net.ResolveUnixAddr(conf.Addr.Network(), conf.Addr.String())
+	if err != nil {
+		return err
+	}
+
+	confCont.Addr = *ua
+	conf.Addr = nil
+	confCont.Plugin = conf
+
+	cache, err := newCache()
+	if err != nil {
+		return err
+	}
+
+	cache.Put(name, confCont)
+
+	return err
+}
+
+func ReadPluginConfigCache(pluginPath string, ref *hplugin.ReattachConfig) error {
+	name := path.Base(pluginPath)
```

Comment:
Why using only the last element of the `pluginPath` as a cache key ? This can easily conflict with an other plugin, for instance `github.com/foo/cli-plugin` and `github.com/bar/cli-plugin` would have the same key right ? 

Using the full plugin path seems safer.

Reply:
that is true, I didn't want to use long strings as the cache key but there is too great of a chance of collisions with the current implementation. I will change to using the full `Path` as the key.